src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -338,27 +338,39 @@ void SQLiteDBEngine::selectData(const std::string& table,
     }
 }
 
-void SQLiteDBEngine::deleteTableRowsData(const std::string& table,
-                                         const nlohmann::json& data)
+void SQLiteDBEngine::deleteTableRowsData(const std::string&    table,
+                                         const nlohmann::json& jsDeletionData)
 {
     if (0 != loadTableData(table))
     {
-        std::vector<std::string> primaryKeyList;
-        if (getPrimaryKeysFromTable(table, primaryKeyList))
+        const auto& itData{ jsDeletionData.find("data")};
+        const auto& itFilter{ jsDeletionData.find("row_filter_opt")};
+
+        if(itData != jsDeletionData.end() && itData->size() > 0)
+        {
+            // Deletion via primary keys on "data" json field.
+            const auto& transaction { m_sqliteFactory->createTransaction(m_sqliteConnection) };
+            deleteRowsbyPK(table, jsDeletionData.at("data"));
```

Comment:
you can use the itData iterator instead of using the at.

Reply:
Done. Good catch